generators/chipyard/src/main/scala/IOBinders.scala

```diff
@@ -260,6 +260,48 @@ class WithSerialTLIOCells extends OverrideIOBinder({
   }).getOrElse((Nil, Nil))
 })
 
+class WithSerialTLAndPassthroughClockPunchthrough extends OverrideLazyIOBinder({
+  (system: CanHavePeripheryTLSerial) => system.serial_tl.map({ serial_io =>
+    implicit val p: Parameters = GetSystemParameters(system)
+
+    val sys = system.asInstanceOf[BaseSubsystem]
+
+    require(p(SerialTLKey).isDefined)
+    val sVal = p(SerialTLKey).get
+
+    // currently only the harness AXI port supports a passthrough clock
+    require(sVal.axiMemOverSerialTLParams.isDefined)
+    val axiDomainParams = sVal.axiMemOverSerialTLParams.get
+
+    val clockSinkNode = axiDomainParams.axiClockParams.map({ clkParams =>
+      // request clock to pass along
+      val node = ClockSinkNode(Seq(ClockSinkParameters(take = Some(ClockParameters(freqMHz = clkParams.clockFreqMHz)))))
+      (node
+        := ClockGroup()(p, ValName("mem_over_serialtl_domain"))
+        := sys.asyncClockGroupsNode)
+      node
+    })
+
+    def clockBundle = clockSinkNode match {
+      case Some(node) => node.in.head._1
+      case None => {
+        val dontCareClockBundle = new ClockBundle(ClockBundleParameters())
+        dontCareClockBundle.clock := DontCare
+        dontCareClockBundle.reset := DontCare
+        dontCareClockBundle
+      }
+    }
+
+    InModuleBody {
+      val port = IO(new SerialAndPassthroughClockResetIO(sVal.width)).suggestName(s"serial_tl_passthrough_clk")
+      port.clocked_serial <> serial_io
+      port.passthrough_clock_reset <> clockBundle
+
+      // return the ports and no IO cells
+      (Seq(port), Nil)
+    }
+  }).getOrElse(InModuleBody{(Nil, Nil)}).asInstanceOf[ModuleValue[IOBinderTuple]]
+})
```

Comment:
Can we avoid punching out the clock here? This breaks the "Chip Config" vs "Platform Config" abstraction, since this IOBinder affects chiptop hardware, but should never appear on an actual chip. 

I realize there's no other easy way to pass the appropriately divided clock out to the harness aximem, but for SW RTL simulation, timing accuracy was never a primary goal anyways. I would much prefer that the RTL simulations use the Harness clock for the sake of preserving this abstraction, even if that is totally unrealistic compared to the chip clock.

Reply:
I'm trying some workarounds for this.

Reply:
Ok sketched out a new implementation. Things to note:

- I still need to fix FireSim to work in a similar fashion
- Is there a nicer way to tell the harness the reference clock frequency of the chip (instead of using `ReferenceClockTracker`)
- Right now there is a difference between the `harness<Clock/Reset>` and the implicit harness clock and reset. I need to fix this...

Reply:
I suggested a way to avoid `ReferenceClockTracker` in another comment.